eabi-toolchain.patch: When using the new OpenMoko toolchain for EABI, u-boot
  (http://wiki.openmoko.org/wiki/Toolchain) fails to build due to a number of
  small compatibility issues. This patch fixes them in a way that is
  backwards-compatible with our ancient OABI toolchain.

  There also seems to be a "competing" patch for OE:
  http://bugzilla.openmoko.org/cgi-bin/bugzilla/show_bug.cgi?id=180

Avoid "warning: target CPU does not support interworking" warnings:
- cpu/arm920t/config.mk (PLATFORM_CPPFLAGS): changed -march=armv4 to
  -march=armv4t
- board/neo1973/gta01/Makefile, board/neo1973/gta02/Makefile: changed
  -march=armv4 to -march=armv4t

Don't hardcode apcs ABI for arm920t:
- cpu/arm920t/config.mk (PLATFORM_CPPFLAGS): removed -mapcs-32 and
  -mabi=apcs-gnu from

__div0 assumes we have some sort of libc:
- cpu/arm920t/config.mk (PLATFORM_LDFLAGS):
  Add dummy "raise" symbol required by __div0 from lib1funcs.asm

Index: u-boot/cpu/arm920t/config.mk
===================================================================
--- u-boot.orig/cpu/arm920t/config.mk
+++ u-boot/cpu/arm920t/config.mk
@@ -24,11 +24,12 @@
 PLATFORM_RELFLAGS += -fno-strict-aliasing  -fno-common -ffixed-r8 \
 	-msoft-float
 
-PLATFORM_CPPFLAGS += -march=armv4
+PLATFORM_CPPFLAGS += -march=armv4t
 # =========================================================================
 #
 # Supply options according to compiler version
 #
 # =========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
+#PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
+PLATFORM_LDFLAGS += --defsym raise=hang
Index: u-boot/board/neo1973/gta01/Makefile
===================================================================
--- u-boot.orig/board/neo1973/gta01/Makefile
+++ u-boot/board/neo1973/gta01/Makefile
@@ -36,7 +36,7 @@
 	$(AR) crv $@ $(OBJS) $(SOBJS)
 
 lowlevel_foo.o:	../common/lowlevel_foo.S
-	$(CC) -c -DTEXT_BASE=0x33F80000 -march=armv4 \
+	$(CC) -c -DTEXT_BASE=0x33F80000 -march=armv4t \
 	  -o lowlevel_foo.o ../common/lowlevel_foo.S
 
 lowlevel_foo:	lowlevel_foo.o ../common/lowlevel_init.o ../common/lowlevel_foo.lds
Index: u-boot/board/neo1973/gta02/Makefile
===================================================================
--- u-boot.orig/board/neo1973/gta02/Makefile
+++ u-boot/board/neo1973/gta02/Makefile
@@ -38,7 +38,7 @@
 	$(AR) crv $@ $(OBJS) $(SOBJS)
 
 lowlevel_foo.o:	../common/lowlevel_foo.S
-	$(CC) -c -DTEXT_BASE=0x33F80000 -march=armv4 \
+	$(CC) -c -DTEXT_BASE=0x33F80000 -march=armv4t \
 	  -o lowlevel_foo.o ../common/lowlevel_foo.S
 
 lowlevel_foo:	lowlevel_foo.o ../common/lowlevel_init.o ../common/lowlevel_foo.lds
